railties/test/application/rake_test.rb

```diff
@@ -141,8 +141,57 @@ def test_rake_routes_with_controller_environment
         end
       RUBY
 
-      ENV['CONTROLLER'] = 'cart'
-      output = Dir.chdir(app_path){ `bin/rails routes` }
+      output = Dir.chdir(app_path){ `bin/rake routes CONTROLLER=cart` }
+      assert_equal "Passing `CONTROLLER` to `bin/rake routes` is deprecated and will be removed in Rails 5.1.\nPlease use `bin/rake routes -c controller_name` instead.\n    Prefix Verb URI Pattern           Controller#Action\n      cart GET  /cart(.:format)       cart#show\nbasketball GET  /basketball(.:format) basketball#index\n", output
```

Comment:
They must. Let's find a way to use `capture` like we do here: https://github.com/rails/rails/blob/a00c36feea6c0271b5ad48a949ef294514fdef52/railties/test/application/test_runner_test.rb#L36 :smile: